vroxy.py

```
@@ -110,117 +19,55 @@ async def get(self):
-                print(f"[{rid}] Cache hit")
-                print(item.resolved_url)
-                print(
-                    f"[{rid}] {item.resolved_format} expires in {item.expiry - curTime} seconds", flush=True
-                )
-                item.lastAccess = curTime
-                return item.resolved_url or ""
-
-        cache_map[_id] = item = Item(url, sort)
-
-        # wait for an pool slot to open
-        timeout = curTime + 30  # 30 seconds timelimit for waiting
-        while pool.count >= pool_max:
-            if curTime > timeout: return None
-            await sleep(1)
-        with YoutubeDL(ytdl_opts) as ytdl:
-            print(f"[{rid}] Resolving '{cacheId}' for url: {url}")
-            print(f"[{rid}] Fetching fresh info", flush=True)
-            pool.add()
-            result = ytdl.extract_info(url, download=False)
-            # print(result.keys())
-            item.resolve(result)
-            pool.remove()
-            print(f"[{rid}] {item.resolved_url}")
-            print(f"[{rid}] {item.resolved_format} expires in {item.expiry - curTime} seconds", flush=True)
-
-        item.lastAccess = curTime
-        return item.resolved_url or ""
-
+        url = None
+        try:
+            url = await resolveUrl(self.request.query)
+            res = web.Response(status=307, headers={"Location": url})
+        except Error400BadRequest:
+            res = web.Response(status=400)
+        except Error403Forbidden:
+            res = web.Response(status=403)
+        except Error403Whitelist:
+            res = web.Response(status=403, text="Domain not in whitelist")
+        except Error404NotFound:
+            res = web.Response(status=404)
+        except Error408Timeout:
+            res = web.Response(status=408)
+        except Error410Gone:
+            res = web.Response(status=410)
+        except Error429TooManyRequests:
+            res = web.Response(status=429)
+        except:
```

This bare except here is too broad and will cause problems with ctrl+c https://peps.python.org/pep-0008/#:~:text=A%20bare%20except%3A%20clause%20will%20catch%20SystemExit%20and%20KeyboardInterrupt%20exceptions